Always set the text and bufsize arguments

almet opened the pull request.

apyrgio commented:
Oh, there's another typo btw:

```diff
diff --git a/dangerzone/capture_output.py b/dangerzone/capture_output.py
index 02118b8d..36207323 100644
--- a/dangerzone/capture_output.py
+++ b/dangerzone/capture_output.py
@@ -35,6 +35,7 @@ class PatchedPopen(original_subprocess_popen):
         """
         stdout = kwargs.get("stdout")
         stderr = kwargs.get("stderr")
+        encoding = kwargs.get("encoding")
         log.debug(f"Running: {shlex.join(args[0])}")
         if stdout is not None or stderr is not None:
             super().__init__(*args, **kwargs)
@@ -43,7 +44,7 @@ class PatchedPopen(original_subprocess_popen):
         # Read the stdout and stderr as streams (text=True and bufsize=1)
         kwargs["stdout"] = subprocess.PIPE
         kwargs["stderr"] = subprocess.PIPE
-        if "encoding" is not None:
+        if encoding is not None:
             kwargs["text"] = True
             kwargs["bufsize"] = 1
```

almet commented:
:scream: Oh wow, this PR was containing a lot of errors. Sorry about that. For context, the research and fix was done on my Windows machine, in which I don't have GitHub credentials.

I copied manually my fixes to my main machine, and did so because the edits were "just a few lines". But clearly it didn't get the attention it deserves and introduced two typos.

Anyway: I've updated the current branch and I think we should be good to go. Funnily enough, the correct answer was yet another thing:

```diff
diff --git a/dangerzone/capture_output.py b/dangerzone/capture_output.py
index 02118b8d..36207323 100644
--- a/dangerzone/capture_output.py
+++ b/dangerzone/capture_output.py
@@ -35,6 +35,7 @@ class PatchedPopen(original_subprocess_popen):
         """
         stdout = kwargs.get("stdout")
         stderr = kwargs.get("stderr")
+        encoding = kwargs.get("encoding")
         log.debug(f"Running: {shlex.join(args[0])}")
         if stdout is not None or stderr is not None:
             super().__init__(*args, **kwargs)
@@ -43,7 +44,7 @@ class PatchedPopen(original_subprocess_popen):
         # Read the stdout and stderr as streams (text=True and bufsize=1)
         kwargs["stdout"] = subprocess.PIPE
         kwargs["stderr"] = subprocess.PIPE
-        if "encoding" is not None:
+        if encoding is None:
             kwargs["text"] = True
             kwargs["bufsize"] = 1
```

---

About the other issue you mention (empty log lines). I have the following questions:

- Were you running your tests on top of these changes?
- I've failed to find the encoding that podman on windows uses (in their docs). Do you have any information about this?
- Do you have any way to reproduce this? I'm happy to have a longer look.

I suspect this to be yet another encoding issue, where some chars are converted to newlines where they shouldn't be.

apyrgio commented:
So, we did a live demo, and in this demo I had removed the `if encoding is None` part. I think it worked fine, so I propose we keep the `text` / `bufsize` flags, regardless of the encoding.
